File: TsogosunProfileAdminWeb/src/app/model/profile-admin/application.ts
import { ApplicationSection } from "./applicationSection";
import { Connection } from "./connection";
import { UserPermission } from "./userPermission";

export interface Application {
    applicationID: number;
    applicationName?: string | undefined;
    isMenuItem: boolean;
    websiteRoute?: string | undefined;
    isActive: boolean;
    applicationSections?: ApplicationSection[] | undefined;
    userPermissions?: UserPermission[] | undefined;
    connections?: Connection[] | undefined;
    status?: string | undefined;
}

File: TsogosunProfileAdmin/MSProfileAdmin/Model/Application.cs

using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace tsogosun.com.MSProfileAdmin.Model
{

    [Table("lApplication")]
    public class Application
    {
        public Application()
        {
            this.ApplicationSections = new HashSet<ApplicationSection>();
            this.UserPermissions = new HashSet<UserPermission>();
            this.Connections = new HashSet<Connection>();
        }
        [Key]
        public int ApplicationID { get; set; }
        public string ApplicationName { get; set; }
        public bool IsMenuItem { get; set; }
        public string WebsiteRoute { get; set; }
        public bool isActive { get; set; } = true;
        public virtual ICollection<ApplicationSection> ApplicationSections { get; set; }
        public virtual ICollection<UserPermission> UserPermissions { get; set; }
        public virtual ICollection<Connection> Connections { get; set; }
        [NotMapped]
        public string Status
        {
            get
            { return isActive == true ? "Active" : "Not Active"; }
        }
    }
}
